File: ios/link2ur/link2ur/Views/Tasks/TaskFilterView.swift
import SwiftUI

struct TaskFilterView: View {
    @Binding var selectedCategory: String?
    @Binding var selectedCity: String?
    @Environment(\.dismiss) var dismiss
    
    // 任务分类映射 (显示名称 -> 后端值)
    var categories: [(name: String, value: String)] {
        [
            (LocalizationKey.commonAll.localized, ""),
            (LocalizationKey.taskCategoryHousekeeping.localized, "Housekeeping"),
            (LocalizationKey.taskCategoryCampusLife.localized, "Campus Life"),
            (LocalizationKey.taskCategorySecondhandRental.localized, "Second-hand & Rental"),
            (LocalizationKey.taskCategoryErrandRunning.localized, "Errand Running"),
            (LocalizationKey.taskCategorySkillService.localized, "Skill Service"),
            (LocalizationKey.taskCategorySocialHelp.localized, "Social Help"),
            (LocalizationKey.taskCategoryTransportation.localized, "Transportation"),
            (LocalizationKey.taskCategoryPetCare.localized, "Pet Care"),
            (LocalizationKey.taskCategoryLifeConvenience.localized, "Life Convenience"),
            (LocalizationKey.taskCategoryOther.localized, "Other")
        ]
    }
    
    // 英国主要城市列表
    var cities: [String] {
        [LocalizationKey.commonAll.localized, "Online", "London", "Edinburgh", "Manchester", "Birmingham", "Glasgow", "Bristol", "Sheffield", "Leeds", "Nottingham", "Newcastle", "Southampton", "Liverpool", "Cardiff", "Coventry", "Exeter", "Leicester", "York", "Aberdeen", "Bath", "Dundee", "Reading", "St Andrews", "Belfast", "Brighton", "Durham", "Norwich", "Swansea", "Loughborough", "Lancaster", "Warwick", "Cambridge", "Oxford", "Other"]
    }
    
    var body: some View {
        NavigationView {
            Form {
                Section(LocalizationKey.taskFilterCategory.localized) {
                    Picker(LocalizationKey.taskFilterSelectCategory.localized, selection: $selectedCategory) {
                        ForEach(categories, id: \.value) { category in
                            Text(category.name)
                                .tag(category.value.isEmpty ? nil : category.value as String?)
                        }
                    }
                }
                
                Section(LocalizationKey.taskFilterCity.localized) {
                    Picker(LocalizationKey.taskFilterSelectCity.localized, selection: $selectedCity) {
                        ForEach(cities, id: \.self) { city in
                            Text(city == LocalizationKey.commonAll.localized ? LocalizationKey.commonAll.localized : city)
                                .tag(city == "全部" ? nil : city as String?)
                        }
                    }
                }
            }
            .navigationTitle(LocalizationKey.commonFilter.localized)
            .navigationBarTitleDisplayMode(.inline)
            .enableSwipeBack()
            .toolbar {
                ToolbarItem(placement: .navigationBarTrailing) {
                    Button(LocalizationKey.commonDone.localized) {
                        dismiss()
                    }
                }
            }
        }
    }
}

File: link2ur/lib/features/tasks/views/task_filter_view.dart
import 'package:flutter/material.dart';

import '../../../core/design/app_colors.dart';
import '../../../core/design/app_spacing.dart';
import '../../../core/utils/l10n_extension.dart';

/// 任务筛选页
/// 参考iOS TaskFilterView.swift
class TaskFilterView extends StatefulWidget {
  const TaskFilterView({
    super.key,
    this.selectedCategory,
    this.selectedCity,
  });

  final String? selectedCategory;
  final String? selectedCity;

  @override
  State<TaskFilterView> createState() => _TaskFilterViewState();
}

class _TaskFilterViewState extends State<TaskFilterView> {
  String? _selectedCategory;
  String? _selectedCity;

  static const _categories = [
    ('', 'All'),
    ('Housekeeping', 'Housekeeping'),
    ('Campus Life', 'Campus Life'),
    ('Second-hand & Rental', 'Second-hand & Rental'),
    ('Errand Running', 'Errand Running'),
    ('Skill Service', 'Skill Service'),
    ('Social Help', 'Social Help'),
    ('Transportation', 'Transportation'),
    ('Pet Care', 'Pet Care'),
    ('Life Convenience', 'Life Convenience'),
    ('Other', 'Other'),
  ];

  static const _cities = [
    'All', 'Online', 'London', 'Edinburgh', 'Manchester',
    'Birmingham', 'Glasgow', 'Bristol', 'Sheffield', 'Leeds',
    'Nottingham', 'Newcastle', 'Southampton', 'Liverpool',
    'Cardiff', 'Coventry', 'Exeter', 'Leicester', 'York',
    'Aberdeen', 'Bath', 'Dundee', 'Reading', 'St Andrews',
    'Belfast', 'Brighton', 'Durham', 'Norwich', 'Swansea',
    'Loughborough', 'Lancaster', 'Warwick', 'Cambridge',
    'Oxford', 'Other',
  ];

  @override
  void initState() {
    super.initState();
    _selectedCategory = widget.selectedCategory;
    _selectedCity = widget.selectedCity;
  }

  @override
  Widget build(BuildContext context) {
    final l10n = context.l10n;

    return Scaffold(
      appBar: AppBar(
        title: Text(l10n.commonFilter),
        actions: [
          TextButton(
            onPressed: () {
              Navigator.of(context).pop({
                'category': _selectedCategory,
                'city': _selectedCity,
              });
            },
            child: Text(l10n.commonDone),
          ),
        ],
      ),
      body: ListView(
        children: [
          // 分类
          Padding(
            padding: const EdgeInsets.all(AppSpacing.md),
            child: Text(
              l10n.taskFilterCategory,
              style: const TextStyle(
                  fontSize: 16, fontWeight: FontWeight.w600),
            ),
          ),
          ..._categories.map((cat) {
            final value = cat.$1.isEmpty ? null : cat.$1;
            final isSelected = _selectedCategory == value;
            return ListTile(
              title: Text(cat.$2),
              trailing: isSelected
                  ? const Icon(Icons.check, color: AppColors.primary)
                  : null,
              onTap: () => setState(() => _selectedCategory = value),
            );
          }),

          const Divider(),

          // 城市
          Padding(
            padding: const EdgeInsets.all(AppSpacing.md),
            child: Text(
              l10n.taskFilterCity,
              style: const TextStyle(
                  fontSize: 16, fontWeight: FontWeight.w600),
            ),
          ),
          ..._cities.map((city) {
            final value = city == 'All' ? null : city;
            final isSelected = _selectedCity == value;
            return ListTile(
              title: Text(city),
              trailing: isSelected
                  ? const Icon(Icons.check, color: AppColors.primary)
                  : null,
              onTap: () => setState(() => _selectedCity = value),
            );
          }),
        ],
      ),
    );
  }
}
